### ISSUE TITLE
Modern Configure DTD / FPI is used inconsistently

### ISSUE THREAD

**Issue by joakime** [eclipse/jetty.project#3751]
The real, up to date, `configure_9_3.dtd` reference is ...

``` xml
<!DOCTYPE Configure PUBLIC "-//Jetty//Configure//EN" "http://www.eclipse.org/jetty/configure_9_3.dtd">
```

Generally, we are pretty good at using this form.
Yet, about 30% of the time, we have the following alternate usage in our codebase (including documentation).

``` xml
<!DOCTYPE Configure PUBLIC "-//Mort Bay Consulting//DTD Configure//EN" "http://www.eclipse.org/jetty/configure_9_3.dtd">
```

This is a historical anomaly from Jetty 7/8 days.
The FPI back then was `-//Mort Bay Consulting//DTD Configure//EN`, but since Jetty 9.0.0 the FPI was changed to `-//Jetty//Configure//EN`

While technically, **our** XmlConfiguration will use either/or, other tooling will not.

I want to change our usages to be up to date in both our code, and the documentation.

Also of note, @sbordet has pointed out that our DTD URI References are using `http://www.eclipse.org/jetty/configure.dtd` and should be updated to use `https://` now, as we are expecting eclipse.org to require `https://` eventually.

**Comment by joakime** [comment]
Once concern about updating the `http` to `https` references, is that if a user starts using `https` on a version of Jetty that doesn't have that entry in the `XmlConfiguration` then they will be forced to fetch the DTD over the internet.

In other words, using `https` is not backwards compatible to earlier versions of Jetty 9.x.x

I would, however, like to get this change to support `https` in for now, but not update our codebase to use `https` until Jetty 10.x.x

**Comment by joakime** [comment]
Opened PR #3752

**Comment by sbordet** [comment]
I have merged #3752 into `jetty-10.0.x` and also updated (only in `jetty-10.0.x`) the scheme from `http` to `https`.
